scripts/2_create.sql

```diff
@@ -25,37 +25,40 @@ CREATE TABLE IF NOT EXISTS aws_hostnames
 
 -- Notice "PARTITION BY" below.  We're using built-in partitioning from Postgres 10+
 -- See https://blog.timescale.com/scaling-partitioning-data-postgresql-10-explained-cd48a712a9a1/
--- Postgres 11 has many updates for partitioning, and when we can use Postgres 11, we can uncomment
--- the FOREIGN KEY bits.  See:  https://pgdash.io/blog/partition-postgres-11.html
+-- Postgres 11 has many updates for partitioning (see https://pgdash.io/blog/partition-postgres-11.html)
+-- that we are taking advantage of, such as automatic index propagation to child tables.
 
 CREATE TABLE IF NOT EXISTS aws_events_ips_hostnames
 (
     ts TIMESTAMP NOT NULL,
     is_public BOOLEAN NOT NULL,
     is_join BOOLEAN NOT NULL,
     aws_resources_id VARCHAR NOT NULL,
-    -- FOREIGN KEY (aws_resources_id) REFERENCES aws_resources (id),
+    FOREIGN KEY (aws_resources_id) REFERENCES aws_resources (id),
     aws_ips_ip INET NOT NULL,
-    -- FOREIGN KEY (aws_ips_ip) REFERENCES aws_ips (ip),
-    aws_hostnames_hostname VARCHAR
-    -- ,
-    -- FOREIGN KEY (aws_hostnames_hostname) REFERENCES aws_hostnames (hostname)
+    FOREIGN KEY (aws_ips_ip) REFERENCES aws_ips (ip),
+    aws_hostnames_hostname VARCHAR,
+    FOREIGN KEY (aws_hostnames_hostname) REFERENCES aws_hostnames (hostname)
 )
 PARTITION BY
     RANGE
 (
         ts
 );
 
--- And we'll make sure there's an index right away:
--- Postgres 11 has many updates for partitioning, and when we can use Postgres 11, we can uncomment
--- the FOREIGN KEY bits.  See:  https://pgdash.io/blog/partition-postgres-11.html
+-- And we'll make sure there's an index right away that Postgres 11 will automatically propagate to child tables:
+-- (We use aws_ips_ip and aws_hostnames_hostname and timestamp as it is expected these will be the most frequently used in queries)
 
--- CREATE INDEX
--- IF NOT EXISTS aws_events_ips_hostnames_aws_ips_ip_ts_idx ON aws_events_ips_hostnames USING BTREE
--- (aws_ips_ip, ts);
+CREATE INDEX
+IF NOT EXISTS aws_events_ips_hostnames_aws_ips_ip_ts_idx ON aws_events_ips_hostnames USING BTREE
+(aws_ips_ip, ts);
+
+CREATE INDEX
+IF NOT EXISTS aws_events_ips_hostnames_aws_hostnames_hostname_ts_idx ON aws_events_ips_hostnames USING BTREE
+(aws_hostnames_hostname, ts);
```

Comment:
Suggestion: There are too many secondary indexes on table that should be all foreign keys. This seems to me like we're not querying the data correctly. For example, we should not be querying the event table by hostname. Instead, we should query the hostname table by hostname and join it to the events table to get the data we need. The same is true for IPs or any other dimension we query by.

I'll take another look at the schema to see what I've missed but we're treating this table like a fully denormalized view of the data. That's not the original intent, IIRC.